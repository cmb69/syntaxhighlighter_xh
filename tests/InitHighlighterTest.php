<?php

/**
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Syntaxhighlighter_XH.
 *
 * Syntaxhighlighter_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Syntaxhighlighter_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Syntaxhighlighter_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Syntaxhighlighter;

use PHPUnit\Framework\TestCase;
use ApprovalTests\Approvals;

class InitHighlighterTest extends TestCase
{
    public function testIt(): void
    {
        $plugin_cf = XH_includeVar("./config/config.php", 'plugin_cf');
        $conf = $plugin_cf['syntaxhighlighter'];
        $plugin_tx = XH_includeVar("./languages/en.php", 'plugin_tx');
        $lang = $plugin_tx['syntaxhighlighter'];
        $sut = new InitHighlighter("./", $conf, $lang);
        $response = $sut();
        Approvals::verifyHtml($response->hjs());
    }
}
