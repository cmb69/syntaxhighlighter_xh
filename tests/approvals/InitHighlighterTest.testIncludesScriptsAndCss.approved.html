<meta name="syntaxhighlighter.brushes" content='[["applescript","./lib/scripts/shBrushAppleScript.js"],["actionscript3","as3","./lib/scripts/shBrushAS3.js"],["bash","shell","./lib/scripts/shBrushBash.js"],["coldfusion","cf","./lib/scripts/shBrushColdFusion.js"],["cpp","c","./lib/scripts/shBrushCpp.js"],["c#","c-sharp","csharp","./lib/scripts/shBrushCSharp.js"],["css","./lib/scripts/shBrushCss.js"],["delphi","pascal","pas","./lib/scripts/shBrushDelphi.js"],["diff","./lib/scripts/shBrushDiff.js"],["patch","./lib/scripts/shBrushPatch.js"],["erl","erlang","./lib/scripts/shBrushErlang.js"],["groovy","./lib/scripts/shBrushGroovy.js"],["java","./lib/scripts/shBrushJava.js"],["jfx","javafx","./lib/scripts/shBrushJavaFX.js"],["js","jscript","javascript","./lib/scripts/shBrushJScript.js"],["perl","pl","./lib/scripts/shBrushPerl.js"],["php","./lib/scripts/shBrushPhp.js"],["powershell","./lib/scripts/shBrushPowershell.js"],["text","plain","./lib/scripts/shBrushPlain.js"],["py","python","./lib/scripts/shBrushPython.js"],["ruby","rails","ror","rb","./lib/scripts/shBrushRuby.js"],["sass","scss","./lib/scripts/shBrushSass.js"],["scala","./lib/scripts/shBrushScala.js"],["sql","./lib/scripts/shBrushSql.js"],["vb","vbnet","./lib/scripts/shBrushVb.js"],["xml","xhtml","xslt","html","./lib/scripts/shBrushXml.js"]]'>
<meta name="syntaxhighlighter.config" content='{"auto-links":true,"class-name":"","collapse":false,"first-line":1,"gutter":true,"highlight":null,"html-script":false,"smart-tabs":true,"tab-size":4,"toolbar":true}'>
<script type="text/javascript" src="./lib/scripts/shCore.js"></script>
<script type="text/javascript" src="./lib/scripts/shAutoloader.js"></script>
<script type="text/javascript" src="./lib/scripts/shBrushXml.js"></script>
<link rel="stylesheet" href="./lib/styles/shCoreDefault.css" type="text/css">
<link rel="stylesheet" href="./lib/styles/shThemeDefault.css" type="text/css">
<meta name="syntaxhighlighter.strings" content='{"expandSource":"+ expand source","help":"?","alert":"SyntaxHighlighter\n\n","noBrush":"Can\u0027t find brush for:","brushNotHtmlScript":"Brush wasn\u0027t made for html-script option:"}'>
<script type="text/javascript" src="./syntaxhighlighter.min.js"></script>
